src/formio/validators/utils.js

```diff
@@ -1,39 +1,29 @@
 import {parseISO} from 'date-fns';
 
-export const validateBoundaries = (componentType, minBoundary, maxBoundary, value) => {
-  const parsedMinBoundary = minBoundary ? new Date(minBoundary) : null;
-  const parsedMaxBoundary = maxBoundary ? new Date(maxBoundary) : null;
+/**
+ * Validate a date or datetime value against min and max boundaries.
+ * This function can be used for both the date and datetime component,
+ * as the error keys are the same for both.
+ */
+export const validateBoundaries = (minBoundary, maxBoundary, value) => {
+  const parsedMinBoundary = minBoundary ? parseISO(minBoundary) : null;
+  const parsedMaxBoundary = maxBoundary ? parseISO(maxBoundary) : null;
 
   if (!parsedMinBoundary && !parsedMaxBoundary) {
     return {isValid: true};
   }
 
   const parsedValue = parseISO(value);
 
-  let errorKeyMinValue, errorKeyMaxValue;
-  if (componentType === 'date') {
-    errorKeyMinValue = 'minDate';
-    errorKeyMaxValue = 'maxDate';
-  } else if (componentType === 'datetime') {
-    errorKeyMinValue = 'minDatetime';
-    errorKeyMaxValue = 'maxDatetime';
-  }
```

Comment:
In the context of my own ticket I'll need this fix, but I'm also trying to track down which error keys were actually supposed to be used.

Reply:
For the `datetime` component, this part of the code would use `minDatetime` and `maxDatetime` as error keys, but we saw in the actual `MinMaxDatetimeValidator` that we expect `minDate` and `maxDate` to be used. This is the main reason the custom translations were not being picked up

Reply:
Okay, I had to do some serious digging, but I found:

* for the `date` component, we exposed the `minDate` and `maxDate` error keys
* for the `datetime` component, we never deliberately exposed those error keys - it came accidentally with the new form builder. The `datetime` component is the built in component from formio, and uses the `minDate` and `maxDate` error keys
* the time component is out of scope here now.

The `minDatetime` and `maxDatetime` error keys never seem to have been a thing. I found commit 99d984275e27283aa460bf84670b89c1b421320e where the datetime min/max validation behaviour was added, which refers to ticket open-formulieren/open-forms#3755 The associated backend PR never specified those errors keys either, so I think it's okay to just stick to `minDate` and `maxDate` for both component types.